pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5550,204 +5550,168 @@ func TestContinueOnError(t *testing.T) {
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgB:m:typB::failing"), snap.Resources[3].URN)
 }
 
-// TestStackOutputsProgramError tests that previous stack outputs aren't deleted when an update fails because
-// of a program error.
-func TestStackOutputsProgramError(t *testing.T) {
+func TestUpContinueOnErrorCreate(t *testing.T) {
 	t.Parallel()
 
-	var step int
-
 	loaders := []*deploytest.ProviderLoader{
 		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
 			return &deploytest.Provider{}, nil
-		}),
+		}, deploytest.WithoutGrpc),
+		deploytest.NewProviderLoader("pkgB", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				CreateF: func(urn resource.URN, news resource.PropertyMap, timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return "", nil, resource.StatusOK, errors.New("intentionally failed create")
+				},
+			}, nil
+		}, deploytest.WithoutGrpc),
 	}
 
 	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
-		stackURN, _, _, _, err := monitor.RegisterResource(resource.RootStackType, "test", false)
+		failing, _, _, _, err := monitor.RegisterResource("pkgB:m:typB", "failing", true, deploytest.ResourceOptions{
```

Comment:
Should we update RegisterResource here to return the Reason field? Maybe join all the fields into a struct at this point rather than adding another return value, could struct-ify in another PR to keep changes small then just rebase and add Reason to the struct in this PR. I don't like that there's a critical response being returned here that we're not even looking at.

Reply:
Yeah that sounds like a good idea: done in https://github.com/pulumi/pulumi/pull/15988.

Reply:
Done now. I've rebased this PR and temporarily based this PR on that branch to make reviewing easier until that can go through the merge queue.

Reply:
The PR merged now, so this is again based on `master`.